doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1522,144 +1471,107 @@ Managing the local context
+   creates a new subgoal, which becomes the first in the list of subgoals
+   remaining to prove.  (This corresponds to the cut rule of sequent calculus.)
 
-   .. tacv:: assert @type by @tactic
+   The second form adds a new hypothesis with a name chosen by Coq
+   or specified by :n:`@as_ipat`.  If :n:`@ltac_expr3` is specified, it is
+   applied to the new subgoal.
 
-      This tactic behaves like :tacn:`assert` but applies tactic to solve the
-      subgoals generated by assert.
+   :n:`@as_ipat`
+     Gives the name for the hypothesis or specifies how to split it into parts.
+     See :ref:`intropatterns`.
 
-      .. exn:: Proof is not complete.
-         :name: Proof is not complete. (assert)
-         :undocumented:
+   :n:`by @ltac_expr3`
+     Applies :n:`@ltac_expr3` to solve the subgoal generated by :n:`assert`.
 
-   .. tacv:: assert @type as @simple_intropattern
+     If not specified, the tactic applies `exact @term` by default where
+     :token:`type` is the type of :token:`term` and also equivalent to
+     :tacn:`pose proof`.  If the head of term is :token:`ident`, the tactic
+     is equivalent to :tacn:`specialize`.
 
-      If :n:`simple_intropattern` is an intro pattern (see :ref:`intropatterns`),
-      the hypothesis is named after this introduction pattern (in particular, if
-      :n:`simple_intropattern` is :n:`@ident`, the tactic behaves like
-      :n:`assert (@ident : @type)`). If :n:`simple_intropattern` is an action
-      introduction pattern, the tactic behaves like :n:`assert @type` followed by
-      the action done by this introduction pattern.
-
-   .. tacv:: assert @type as @simple_intropattern by @tactic
-
-      This combines the two previous variants of :tacn:`assert`.
+   .. exn:: Not a proposition or a type.
 
-   .. tacv:: assert (@ident := @term)
+      Arises when the argument :token:`type` is neither of type :g:`Prop`,
+      :g:`Set` nor :g:`Type`.
 
-      This behaves as :n:`assert (@ident : @type) by exact @term` where
-      :token:`type` is the type of :token:`term`. This is equivalent to using
-      :tacn:`pose proof`. If the head of term is :token:`ident`, the tactic
-      behaves as :tacn:`specialize`.
+   .. exn:: Proof is not complete.
+      :name: Proof is not complete. (assert)
+      :undocumented:
 
-      .. exn:: Variable @ident is already declared.
-         :undocumented:
+   .. exn:: Variable @ident is already declared.
+      :undocumented:
 
-.. tacv:: eassert @type as @simple_intropattern by @tactic
-   :name: eassert
+.. tacn:: eassert ( @ident : @term ) {? by @ltac_expr3 }
+          eassert @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: eassert; _
 
    While the different variants of :tacn:`assert` expect that no existential
    variables are generated by the tactic, :tacn:`eassert` removes this constraint.
    This lets you avoid specifying the asserted statement completely before starting
```

Comment:
Should we also mention that the existential variables introduced can be found as shelved goals?

Reply:
I added a hyperlink to the definition of "existential variable".  To [here](https://coq.inria.fr/distrib/current/refman/language/extensions/evars.html#term-existential-variable).

AFAICT, the only way to see the list of existential variables is with "Show Existentials" (mentioned at the link).  The link should probably mention some other obvious operations that seem to be missing (or I just don't recall them).
- Is there a tactic to shelve or unshelve a specific item by name?
- It's odd to have an `Unshelve` command; it seems like it should be `unshelve` with no tactic name

Reply:
I believe existentials with user given names can be accessed directly with the goal selectory. So you could `Unshelve` all the goals and then `shelve` the existential you want using the goal selector. I am not 100% certain how to do it however.